src/Bicep.Core/Analyzers/Linter/Rules/EnvironmentUrlHardcodedRule.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Configuration;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class EnvironmentUrlHardcodedRule : LinterRuleBase
+    {
+        public new const string Code = "no-hardcoded-env-urls";
```

Comment:
>Code [](http://example.com/codeflow?start=32&length=4)

nit - This pattern actually hurts readability. My recommendation would be to inline the constant in the base constructor invocation. If the value is needed, it can be accessed via the `this.Code` since the member is public.

This applies to all the rules using the pattern.

Reply:
I'd say these tests shouldn't use the constants and actually be hardcoded. Otherwise, an accidental rename will make the tests pass and miss what is a breaking change. These constants shouldn't change pretty much ever, so there isn't much upside from reuse.

Reply:
Okay, I buy that as far as tests go.  Will change after merge.